src/main/scala/com/fulcrumgenomics/vcf/api/VcfConversions.scala

```diff
@@ -0,0 +1,330 @@
+  }
+
+  /** Converts back from the scala [[VcfCount]] back to the information needed to specify the count
+    * in HTSJDK.
+    *
+    * @param in the [[VcfCount]] representing how many values an INFO or FORMAT field should have
+    * @return either a [[VCFHeaderLineCount]] when possible or an [[Int]] if the count is [[Fixed]]
+    */
+  def toJavaCount(in: VcfCount): Either[VCFHeaderLineCount, Int] = {
+    in match {
+      case VcfCount.OnePerAltAllele    => Left(VCFHeaderLineCount.A)
+      case VcfCount.OnePerGenotype     => Left(VCFHeaderLineCount.G)
+      case VcfCount.OnePerAllele       => Left(VCFHeaderLineCount.R)
+      case VcfCount.Unknown            => Left(VCFHeaderLineCount.UNBOUNDED)
+      case VcfCount.Fixed(n)           => Right(n)
+    }
+  }
+
+  /**
+    * Converts back from the scala [[VcfFieldType]] into the java/HTSJDK equivalent.
+    */
+  def toJavaKind(in: VcfFieldType ): VCFHeaderLineType = in match {
+    case VcfFieldType.Character => VCFHeaderLineType.Character
+    case VcfFieldType.Flag      => VCFHeaderLineType.Flag
+    case VcfFieldType.Float     => VCFHeaderLineType.Float
+    case VcfFieldType.Integer   => VCFHeaderLineType.Integer
+    case VcfFieldType.String    => VCFHeaderLineType.String
+  }
+
+  /**
+    * Converts a [[VariantContext]] and all nested classes into a [[Variant]] and set of [[Genotype]]s.
+    *
+    * @param in the [[VariantContext]] to be converted
+    * @param header the scala [[VcfHeader]] which contains the definitions of all the INFO and FORMAT
+    *               fields as well as the ordered list of sample names.
+    * @return a [[Variant]] instance that is a copy of the [[VariantContext]] and does not rely on it
+    *         post-return
+    */
+  def toScalaVariant(in: VariantContext, header: VcfHeader): Variant = {
+    // Build up the allele set
+    val ref  = Allele(in.getReference.getDisplayString)
+    val alts = in.getAlternateAlleles.map(a => Allele(a.getDisplayString)).toIndexedSeq
+    val alleles = AlleleSet(ref, alts)
+    val alleleMap = alleles.map(a => a.toString -> a).toMap
+
+    // Build up the genotypes
+    val gts = in.getGenotypes.map { g =>
+      val calls = g.getAlleles.map(a => alleleMap(a.getDisplayString)).toIndexedSeq
+      val attrs = {
+        val buffer = new ArrayBuffer[(String,Any)](g.getExtendedAttributes.size() + 4)
+        if (g.hasAD) buffer.append("AD" -> g.getAD.toIndexedSeq)
+        if (g.hasDP) buffer.append("DP" -> g.getDP)
+        if (g.hasGQ) buffer.append("GQ" -> g.getGQ)
+        if (g.hasPL) buffer.append("PL" -> g.getPL.toIndexedSeq)
+
+        g.getExtendedAttributes.keySet().foreach { key =>
+          val value = g.getExtendedAttribute(key)
+
+          header.format.get(key) match {
+            case Some(hd) => buffer.append(key -> toTypedValue(value, hd.kind, hd.count))
+            case None     => throw new IllegalStateException(s"Format field $key not described in header.")
```

Comment:
Here and again for the INFO map below, we could decide to do a best-effort conversion if the header line is missing.  I'm thinking we'd basically do the following:
1. Split the value on commas
2. Attempt to parse all values as Int, then Float then String, picking the type where all values are valid
3. If there's a single value, return the single value; if there are multiple return them as an ArrayAttribute

It wouldn't be perfect, but presumably better than blowing up?